Foundation/NSURLSession/NSURLSessionTask.swift

```diff
@@ -833,10 +833,13 @@ extension URLSessionTask {
             guard case .inMemory(let bodyData) = bodyDataDrain else {
                 fatalError("Task has data completion handler, but data drain is not in-memory.")
             }
+
             guard let s = session as? URLSession else { fatalError() }
-           
-            var data = Data(capacity: bodyData!.length)
-            data.append(Data(bytes: bodyData!.bytes, count: bodyData!.length)) 
+
+            var data = Data()
+            if let body = bodyData {
+                data.append(Data(bytes: body.bytes, count: body.length))
+            }
```

Comment:
Shouldn't we be able to use `public init(bytes: UnsafeRawPointer, count: Int)` on Data here?